fastlane/Fastfile

```diff
@@ -99,31 +136,43 @@ in xcode user defined build configuration and just reflected in `Info.plist` dur
 
     desc "Read current version number from project configuration."
     current_version_version = read_project_property(
-      build_configuration: build_configurations[:testflight],
+      build_configuration: build_configurations[build_configuration],
       build_setting: 'CURRENT_VERSION_NUMBER'
     )
 
-    desc "Read current build number from project configuration."
-    current_build_version = read_project_property(
-      build_configuration: build_configurations[:testflight],
-      build_setting: 'CURRENT_BUILD_NUMBER'
+    desc "Read current build number from TestFlight."
+    current_build_version = latest_testflight_build_number(
+      app_identifier: bundle_identifier,
+      version: current_version_version,
+      initial_build_number: first_build
     )
 
     desc "Get the type of release desired"
+    desc "If there's only one bump_type allowed it will use that one"
+    desc "If there isn't a bump_type specified and there are many allowed bump_types, it will ask the user"
+
+    is_first_deploy = current_version_version == first_version
     bump_type = options[:bump_type]
-    while not release_bump_types.include? bump_type do
-      bump_type = UI.input "Choose the `bump_type` representing the type of deploy. It can be any of %s" % release_bump_types.to_s
+
+    if allowed_bump_types.count == 1
+      bump_type = is_first_deploy ? all_bump_types.last : allowed_bump_types.first
```

Comment:
I don't like this. We're adding complicated logic in this function to handle something that should not be handled here.

I think the best we can do is:

```
lane :release_qa do
    // Read current version.
    bump_type = is_first_deploy ? :major : :build
    release build_configuration: :qa bump_type
end
```

And let `release` do something easier. If it receives a `bump_type` and it's valid, use it. Otherwise ask the user until it provides a valid one. Doesn't matter if the amount of valid ones are 0, 1 or 1000.

What do you think?

Reply:
I'd check if Spaceship allows us to check whether there's a version uploaded or not. That'd be even better.

Reply:
I can't find anything in Spaceship to get versions, amount of versions or if there is an existing version.

I'm not sure I agree. If we have another environment that has only one allowed bump_type, we'd have to do the same for that one too, which would be redundant.

Also, this lane already did this check before, the only difference is that now instead of asking the user if it wants major as the first deploy (which it also didn't make much sense as it failed if you didn't agree), it does it automatically when there's only one allowed type. It's transparent to the user whether that happens or not, because in those cases (release_qa for instance) they don't even know which bump_type is being used. In those cases where there are more allowed types (release_testflight), as the user can pass a bump_type parameter, we have to let them know if that parameter is valid or not.